ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import io.confluent.common.config.AbstractConfig;
+import io.confluent.common.config.ConfigDef;
+import io.confluent.common.config.ConfigDef.Importance;
+import io.confluent.common.config.ConfigDef.Type;
+import io.confluent.ksql.util.KsqlConfig;
+import io.vertx.core.json.JsonObject;
+import java.util.Map;
+
+/**
+ * Config for the API server
+ */
+public class ApiServerConfig extends AbstractConfig {
+
+  private static final String PROPERTY_PREFIX = "apiserver.";
+
+  public static final String VERTICLE_INSTANCES = propertyName("verticle-instances");
+  public static final int DEFAULT_VERTICLE_INSTANCES =
+      2 * Runtime.getRuntime().availableProcessors();
+  public static final String VERTICLE_INSTANCES_DOC =
+      "The number of server verticle instances to start. Usually you want at least many instances"
+          + " as there are cores you want to use, as each instance is single threaded.";
+
+  public static final String LISTEN_HOST = propertyName("listen-host");
+  public static final String DEFAULT_LISTEN_HOST = "0.0.0.0";
+  public static final String LISTEN_HOST_DOC =
+      "The hostname to listen on";
+
+  public static final String LISTEN_PORT = propertyName("listen-port");
+  public static final int DEFAULT_LISTEN_PORT = 8089;
+  public static final String LISTEN_PORT_DOC =
+      "The port to listen on";
+
+  public static final String KEY_PATH = propertyName("key-path");
+  public static final String KEY_PATH_DOC =
+      "Path to key file";
+
+  public static final String CERT_PATH = propertyName("cert-path");
+  public static final String CERT_PATH_DOC =
+      "Path to cert file";
+
+  private static String propertyName(final String name) {
+    return KsqlConfig.KSQL_CONFIG_PROPERTY_PREFIX + PROPERTY_PREFIX + name;
+  }
+
+  private static final ConfigDef CONFIG_DEF = new ConfigDef()
+      .define(
+          VERTICLE_INSTANCES,
+          Type.INT,
+          DEFAULT_VERTICLE_INSTANCES,
+          Importance.LOW,
+          VERTICLE_INSTANCES_DOC)
+      .define(
+          LISTEN_HOST,
+          Type.STRING,
+          DEFAULT_LISTEN_HOST,
+          Importance.MEDIUM,
+          LISTEN_HOST_DOC)
+      .define(
+          LISTEN_PORT,
+          Type.INT,
+          DEFAULT_LISTEN_PORT,
+          Importance.MEDIUM,
+          LISTEN_PORT_DOC)
+      .define(
+          KEY_PATH,
+          Type.STRING,
+          null,
+          Importance.MEDIUM,
+          KEY_PATH_DOC)
+      .define(
+          CERT_PATH,
+          Type.STRING,
+          null,
+          Importance.MEDIUM,
+          CERT_PATH_DOC);
```

Comment:
FYI, you can set `Validator`s on these, e.g. to fail early if the port is negative etc.

While on the subject of ports - Jetty supports auto-port assignment if port is zero. Does vert.x have something similar?  I've run into bind exceptions locally when doing multiple things - build servers could run into the same issue...

Reply:
Yes, this is supported by Vert.x https://vertx.io/docs/apidocs/io/vertx/core/http/HttpServer.html#actualPort--